### DISCUSSION TITLE
modem_cellular: How can I skip the periodic_chat_script,  Insted I want to call  custom AT cmd from application space

### DISCUSSION THREAD

**Discussion by srikanthpalvai** [zephyrproject-rtos/zephyr#73188] [Q&A]
zephyr\drivers\modem\modem_cellular.c
const struct modem_chat_script *periodic_chat_script;

**Comment by srikanthpalvai** [comment]
@bjarki-trackunit 
Could you please suggest me

**Reply by bjarki-andreasen** [reply]
Interestingly enough I'm working on this now :) I am adding free to use modem pipes that can be used for shells, custom commands, gnss pass through etc. I will share it later today in a PR :)

It is

**Reply by srikanthpalvai** [reply]
Hi @bjarki-trackunit  Is the PR Up for the review? if not Please share the patch, So I try them locally

**Reply by bjarki-andreasen** [reply]
Not yet, I will mention it when its up, you can also follow the discord zephyr modem channel where these things are also posted/discussed

**Comment by srikanthpalvai** [comment]
Thanks for the update, @bjarki-trackunit  That sounds like a great addition. Could you please share more details about the modem pipes and how they can be used for custom commands? Looking forward to your PR.

Few Suggestions from my end:
I think it's better to come up with kconfig flag **SKIP_PERIODIC_CHAT_SCRIPT**.

**Comment by bjarki-andreasen** [comment]
PR for reveiw https://github.com/zephyrproject-rtos/zephyr/pull/73240

**Reply by srikanthpalvai** [reply]
In my opinion, using the shell would be useful when the end user needs such flexibility. However, in my case, I don't use the shell. I want to send custom AT commands such are AT+CSIM  through a wrapper API  and validate the response from zephyr/samples/net/cellular_modem/src/main.c

**Reply by bjarki-andreasen** [reply]
The shell is just a quick sample, you can send anything :) In your case, create a modem_chat instance, create a chat script containing AT commands such as AT+CSIM, attach the modem_chat to the pipe exposed by the pipelink (like the sample), and you are good to go :)

**Reply by srikanthpalvai** [reply]
Thank you

**Comment by srikanthpalvai** [comment]
@bjarki-trackunit 
I didn't seen anywhere we are skiping the periodic_chat_script from the changes 
Could you please come up with flag  SKIP_PERIODIC_CHAT_SCRIPT=y to define it in the prj.conf

this ticket i have speically for this feature.

**Reply by bjarki-andreasen** [reply]
There is actually a simple way to skip the periodic script, set the timeout to a large value, like 3600000, it will run once every hour :) Disabling it completely could be added, and you are welcome to create a PR to remove it completely as well :)
